mimxrt1060_evk_hyperflash board definition is broken

**Describe the bug**
A problem was introduced with #38676 causing the `mimxrt1060_evk_hyperflash` board definition (and perhaps others) to break.

**To Reproduce**
Steps to reproduce the behavior:
1. `west -v build -b mimxrt1060_evk_hyperflash samples/drivers/flash_shell`
2. See error:

```
ZEPHYR_BASE=/home/brix/Projects/zephyrproject/zephyr (origin: env)
-- west build: generating a build system
cmake version 3.21.3 is OK; minimum version is 3.13.1
Running CMake: /home/brix/Projects/zephyrproject/venv/bin/cmake -DWEST_PYTHON=/home/brix/Projects/zephyrproject/venv/bin/python3 -B/home/brix/Projects/zephyrproject/zephyr/build -S/home/brix/Projects/zephyrproject/zephyr/samples/drivers/flash_shell -GNinja -DBOARD=mimxrt1060_evk_hyperflash
Including boilerplate (Zephyr base): /home/brix/Projects/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/brix/Projects/zephyrproject/zephyr/samples/drivers/flash_shell
-- Zephyr version: 2.7.99 (/home/brix/Projects/zephyrproject/zephyr), build: v2.7.99-917-g71c0cebac904
-- Found Python3: /home/brix/Projects/zephyrproject/venv/bin/python3 (found suitable exact version "3.8.10") found components: Interpreter 
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
-- Board: mimxrt1060_evk_hyperflash
-- Cache files will be written to: /home/brix/.cache/zephyr
-- Using toolchain: zephyr 0.13.1 (/opt/zephyr-sdk)
-- Found dtc: /opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/brix/Projects/zephyrproject/zephyr/boards/arm/mimxrt1060_evk/mimxrt1060_evk_hyperflash.dts
-- Generated zephyr.dts: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/brix/Projects/zephyrproject/zephyr/samples/drivers/flash_shell/Kconfig
Loaded configuration '/home/brix/Projects/zephyrproject/zephyr/boards/arm/mimxrt1060_evk/mimxrt1060_evk_hyperflash_defconfig'
Merged configuration '/home/brix/Projects/zephyrproject/zephyr/samples/drivers/flash_shell/prj.conf'
Configuration saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Error at /home/brix/Projects/zephyrproject/zephyr/cmake/extensions.cmake:2708 (message):
  dt_prop(compat_flash ...) missing required argument: PATH
Call Stack (most recent call first):
  /home/brix/Projects/zephyrproject/zephyr/drivers/flash/CMakeLists.txt:30 (dt_prop)


-- Configuring incomplete, errors occurred!
See also "/home/brix/Projects/zephyrproject/zephyr/build/CMakeFiles/CMakeOutput.log".
See also "/home/brix/Projects/zephyrproject/zephyr/build/CMakeFiles/CMakeError.log".
FATAL ERROR: command exited with status 1: /home/brix/Projects/zephyrproject/venv/bin/cmake -DWEST_PYTHON=/home/brix/Projects/zephyrproject/venv/bin/python3 -B/home/brix/Projects/zephyrproject/zephyr/build -S/home/brix/Projects/zephyrproject/zephyr/samples/drivers/flash_shell -GNinja -DBOARD=mimxrt1060_evk_hyperflash

```
**Expected behavior**
Board definition should not be broken.

**Impact**
This affects CI for other PRs (see e.g. https://buildkite.com/zephyr/zephyr/builds/45183)

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK,
 - Commit SHA: 71c0cebac9049ef2ce5ea377681ac8140a341598

--- a/boards/arm/mimxrt1050_evk/mimxrt1050_evk_qspi.dts
+++ b/boards/arm/mimxrt1050_evk/mimxrt1050_evk_qspi.dts
@@ -10,13 +10,17 @@
 
 / {
 	chosen {
-		/delete-property/ zephyr,flash-controller;
-		/delete-property/ zephyr,flash;
-		/delete-property/ zephyr,code-partition;
+		zephyr,flash-controller = &flexspi;
+		zephyr,flash = &is25wp064;
+		zephyr,code-partition = &slot0_partition;
 	};
 };
 
 &flexspi {
+	status = "okay";
+	ahb-prefetch;
+	ahb-read-addr-opt;
+	rx-clock-source = <1>;
 	reg = <0x402a8000 0x4000>, <0x60000000 DT_SIZE_M(8)>;
 	is25wp064: is25wp064@0 {
 		compatible = "nxp,imx-flexspi-nor";
@@ -26,5 +30,34 @@
 		spi-max-frequency = <133000000>;
 		status = "okay";
 		jedec-id = [9d 70 17];
+		erase-block-size = <4096>;
+		write-block-size = <1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			boot_partition: partition@0 {
+				label = "mcuboot";
+				reg = <0x00000000 DT_SIZE_K(64)>;
+			};
+			slot0_partition: partition@10000 {
+				label = "image-0";
+				reg = <0x00010000 DT_SIZE_M(3)>;
+			};
+			slot1_partition: partition@310000 {
+				label = "image-1";
+				reg = <0x00310000 DT_SIZE_M(3)>;
+			};
+			scratch_partition: partition@610000 {
+				label = "image-scratch";
+				reg = <0x00610000 DT_SIZE_K(128)>;
+			};
+			storage_partition: partition@630000 {
+				label = "storage";
+				reg = <0x00630000 DT_SIZE_K(1856)>;
+			};
+		};
 	};
 };

--- a/boards/arm/mimxrt1060_evk/mimxrt1060_evk_hyperflash.dts
+++ b/boards/arm/mimxrt1060_evk/mimxrt1060_evk_hyperflash.dts
@@ -10,13 +10,21 @@
 
 / {
 	chosen {
-		/delete-property/ zephyr,flash-controller;
-		/delete-property/ zephyr,flash;
-		/delete-property/ zephyr,code-partition;
+		zephyr,flash-controller = &s26ks512s0;
+		zephyr,flash = &s26ks512s0;
+		zephyr,code-partition = &slot0_partition;
 	};
 };
 
 &flexspi {
+	status = "okay";
+	ahb-prefetch;
+	ahb-read-addr-opt;
+	ahb-bufferable;
+	ahb-cacheable;
+	sck-differential-clock;
+	combination-mode;
+	rx-clock-source = <3>;
 	reg = <0x402a8000 0x4000>, <0x60000000 DT_SIZE_M(64)>;
 	s26ks512s0: s26ks512s@0 {
 		compatible = "nxp,imx-flexspi-hyperflash";
@@ -24,6 +32,42 @@
 		label = "S26KS512S";
 		reg = <0>;
 		spi-max-frequency = <166000000>;
+		word-addressable;
+		cs-interval-unit = <1>;
+		cs-interval = <2>;
+		cs-hold-time = <0>;
+		cs-setup-time = <3>;
+		data-valid-time = <1>;
+		column-space = <3>;
+		ahb-write-wait-unit = <2>;
+		ahb-write-wait-interval = <20>;
 		status = "okay";
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			boot_partition: partition@0 {
+				label = "mcuboot";
+				reg = <0x00000000 DT_SIZE_K(256)>;
+			};
+			slot0_partition: partition@40000 {
+				label = "image-0";
+				reg = <0x00040000 DT_SIZE_M(3)>;
+			};
+			slot1_partition: partition@340000 {
+				label = "image-1";
+				reg = <0x00340000 DT_SIZE_M(3)>;
+			};
+			scratch_partition: partition@640000 {
+				label = "image-scratch";
+				reg = <0x00640000 DT_SIZE_K(768)>;
+			};
+			storage_partition: partition@700000 {
+				label = "storage";
+				reg = <0x00700000 DT_SIZE_M(57)>;
+			};
+		};
 	};
 };

